### PR TITLE
Drawer component and Tanstack Routing

### PR DESCRIPTION
**Pull request by Julianna-Venter**
Moved the nav bar into drawer component and added routing via drawer and buttons. This is still hard coded, purely UI (not web responsive yet)

### CODE REVIEW

File: src/Navigation/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import { cibGithub, cilSearch } from "@coreui/icons";
+import CIcon from "@coreui/icons-react";
+import { Link } from "@tanstack/react-router";
+
+interface DrawerProps {
+  username: string;
+}
+
+const Drawer: React.FC<DrawerProps> = ({ username }) => {
```
Comment (Hugovs2000) [inline src/Navigation/Drawer.tsx]: I think you can do  `const Drawer: React.FC<DrawerProps> = ({ username }:{username:string}) => {...}` and then you do not neet the interface.

File: src/commonComponents/BackgroundAssets.tsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
 function BackgroundAssets() {
   return (
     <div className="h-screen w-screen overflow-hidden absolute -z-20">
-      <div className="w-[600px] h-[550px] bg-primary_blue absolute -z-20 rotate-45 rounded-badge left-[-18.5rem] top-[-15rem] flex-0"></div>
-      <div className="w-[300px] h-[150px] border-solid border-12 border-secondary_orange absolute rounded-badge -z-10 left-[6rem] top-[-8rem] rotate-45 flex-0"></div>
-      <div className="w-[600px] h-[550px] bg-primary_blue absolute -z-20 rotate-45 rounded-badge right-[-15rem] bottom-[-25rem] flex-0"></div>
-      <div className="w-[300px] h-[150px] border-solid border-12 border-secondary_orange absolute rounded-badge -z-10 right-[-14rem] bottom-[4rem] rotate-45 flex-0"></div>
+      <div className="w-[37.5rem] h-[34.375rem] bg-primary_blue absolute -z-20 rotate-45 rounded-badge left-[-18.5rem] top-[-15rem] flex-0"></div>
+      <div className="w-[18.75rem] h-[9.375rem] border-solid border-12 border-secondary_orange absolute rounded-badge -z-10 left-[6rem] top-[-8rem] rotate-45 flex-0"></div>
```
Comment (Hugovs2000) [inline src/commonComponents/BackgroundAssets.tsx]: Kinda sketch, looks like you are fighting the browser.

**Review by Hugovs2000 — CHANGES_REQUESTED** [review]
Just the comments about how you are destructing your props and questionable underground browser fighting 👀

File: src/Navigation/Drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+          >
+            <label className="font-semibold">Users</label>
+            {username && (
+              <svg
+                xmlns="http://www.w3.org/2000/svg"
+                fill="none"
+                viewBox="0 0 24 24"
+                strokeWidth={2.8}
+                stroke="currentColor"
+                className="w-4 h-4 text-ligher_text"
```
Comment (yanevdw) [inline src/Navigation/Drawer.tsx]: On ```Line 55```, I assume this should be "lighter"?

File: src/userForm/UserForm.tsx [hunk 4]
```diff
@@ -15,75 +16,80 @@ function UserForm() {
...
-                  setFieldValue("username", value ? value.value : "");
-                }}
-                options={options}
-                value={selectedValue || null}
-                className="w-full"
-                placeholder="Enter a Username"
-              />
+          <div className="flex flex-col gap-1.5">
+            <h1 className="text-dark_text text-2xl lg:text-3xl">Find a User</h1>
+            <h2 className="text-ligher_text text-sm lg:text-base">
```
Comment (yanevdw) [inline src/userForm/UserForm.tsx]: On ```Line 30```: Same typo here.

**Review by yanevdw — CHANGES_REQUESTED** [review]
- As far as the drawer and routing is concerned, nicely done.
- Take a look at the comments.
-  Please take a look at your ```tailwind.config.js``` file, specifically your naming conventions for the custom colours. The naming convention should be kebab-case.
- Also just a suggestion on the naming of the custom text colours, just name the colour insted of ```text-colour```.

**Comment by Julianna-Venter** [comment]
> * As far as the drawer and routing is concerned, nicely done.
> * Take a look at the comments.
> * Please take a look at your `tailwind.config.js` file, specifically your naming conventions for the custom colours. The naming convention should be kebab-case.
> * Also just a suggestion on the naming of the custom text colours, just name the colour insted of `text-colour`.

As for the `tailwind.config.js` file naming conventions for the colors, I did some research on typical naming conventions, and color names are usually lowercase, I do see however that a hyphen is more conventional than an underscore, so I will change that. I used the Keyword Naming convention, which is to name a color based on the context in which that color is used. If I have time I plan on adding a dark theme, that is why I specified `primary_blue` instead of `primary_color`.

**Review by yanevdw — APPROVED** [review]
Looks good

File: src/profilePage/ProfilePage.tsx [hunk 5]
```diff
@@ -12,103 +7,18 @@ import {
...
-          className="w-6 h-6"
-        >
-          <path
-            fillRule="evenodd"
-            d="M6.32 2.577a49.255 49.255 0 0 1 11.36 0c1.497.174 2.57 1.46 2.57 2.93V21a.75.75 0 0 1-1.085.67L12 18.089l-7.165 3.583A.75.75 0 0 1 3.75 21V5.507c0-1.47 1.073-2.756 2.57-2.93Z"
-            clipRule="evenodd"
-          />
-        </svg> </div>*/}
-      </nav>
+      <Drawer username={"julianna-venter"} />
```
Comment (Musa-Mabasa) [inline src/profilePage/ProfilePage.tsx:19]: You do not have to evaluate the attribute assignment for string literals 🎨

**Review by Musa-Mabasa — APPROVED** [review]
Left one comment about attribute assignment. Either than that, everything looked good.

File: src/commonComponents/BackgroundAssets.tsx (hunk 2 above)
Comment (Musa-Mabasa) [reply src/commonComponents/BackgroundAssets.tsx]: I agree, however I think the needed implementation for now is to not scale anything according to the browser.

**Review by Hugovs2000 — APPROVED** [review]
All good

**Pull request merged**